src/Primitives/src/StringSegmentComparer.cs

```diff
@@ -34,8 +35,17 @@ public bool Equals(StringSegment x, StringSegment y)
 
         public int GetHashCode(StringSegment obj)
         {
-            // TODO: PERF; Note that .NET Core strings use randomized hash codes for security reasons.
+#if NETCOREAPP
+            return string.GetHashCode(obj.AsSpan(), Comparison);
+#else
+            if (!obj.HasValue)
+            {
+                return 0;
+            }
```

Comment:
Is this just being done for perf for the degenerate case?

Reply:
The current code throws if you have a `StringSegment` instance without a backing buffer (e.g. `var x = default(StringSegment);`). I used existing test data for `StringSegment` which cover this case.